mila_datamodules/cli/prepare_torchvision.py

```diff
@@ -45,56 +58,78 @@ def __call__(
         """
         Path(root).mkdir(parents=True, exist_ok=True)
 
-        constructor_kwargs = constructor_kwargs.copy()  # type: ignore
+        dataset_kwargs = dataset_kwargs.copy()  # type: ignore
         if "download" in inspect.signature(self.dataset_type).parameters:
-            constructor_kwargs["download"] = True
+            dataset_kwargs["download"] = True
 
         logger.debug(
-            f"Using dataset constructor: {self.dataset_type} with args {constructor_args}, and "
-            f"kwargs {constructor_kwargs}"
+            f"Using dataset constructor: {self.dataset_type} with args {dataset_args}, and "
+            f"kwargs {dataset_kwargs}"
         )
-        dataset_instance = self.dataset_type(str(root), *constructor_args, **constructor_kwargs)
+        dataset_instance = self.dataset_type(str(root), *dataset_args, **dataset_kwargs)
         if is_local_main():
             print(dataset_instance)
         return str(root)
 
 
-class SymlinkArchives(PrepareVisionDataset[VD, P]):
-    """Creates symlinks to the archives in the `root` directory."""
+class SymlinkDatasetFiles(AbstractPrepareVisionDataset[VD, P]):
+    """Creates symlinks to the datasets' files in the `root` directory."""
 
     def __init__(
         self,
-        dataset_type: Callable[Concatenate[str, P], VD],
-        relative_paths_to_archives: dict[str, str | Path],
+        source: str | Path = None,
+        files: dict[str, str | Path] = None
```

Comment:
Use `<type> | None` or `Optional[<type>]` if the default value is `None`, otherwise the type annotation is incorrect.

Reply:
Will do thx!